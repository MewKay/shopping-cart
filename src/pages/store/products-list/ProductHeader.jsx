import PropTypes from "prop-types";
import { Link, useParams } from "react-router-dom";
import styles from "./ProductHeader.module.css";

const ProductHeader = ({ searchedProduct, onSearchChange }) => {
  const { category } = useParams();

  const allFilterActive = category === "all" ? styles["active"] : "";
  const menClothingFilterActive =
    category === "men's clothing" ? styles["active"] : "";
  const womenClothingFilterActive =
    category === "women's clothing" ? styles["active"] : "";
  const jeweleryFilterActive = category === "jewelery" ? styles["active"] : "";
  const electronicsFilterActive =
    category === "electronics" ? styles["active"] : "";

  return (
    <div className={styles["products-list-header"]}>
      <ul className={styles["category-filter"]}>
        <li>
          <Link to={"/store/all"} className={allFilterActive}>
            All
          </Link>
        </li>
        <li>
          <Link
            to={"/store/men's clothing"}
            className={menClothingFilterActive}
          >
            Men&apos;s Clothing
          </Link>
        </li>
        <li>
          <Link
            to={"/store/women's clothing"}
            className={womenClothingFilterActive}
          >
            Women&apos;s Clothing
          </Link>
        </li>
        <li>
          <Link to={"/store/jewelery"} className={jeweleryFilterActive}>
            Jewelery
          </Link>
        </li>
        <li>
          <Link to={"/store/electronics"} className={electronicsFilterActive}>
            Electronics
          </Link>
        </li>
      </ul>
      <input
        className={styles["search-product"]}
        type="text"
        placeholder="Search for a product..."
        value={searchedProduct}
        onChange={onSearchChange}
      />
    </div>
  );
};

ProductHeader.propTypes = {
  searchedProduct: PropTypes.string.isRequired,
  onSearchChange: PropTypes.func.isRequired,
};

export default ProductHeader;
