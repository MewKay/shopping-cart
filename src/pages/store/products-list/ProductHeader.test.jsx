import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ProductHeader from "./ProductHeader";

describe("ProductHeader", () => {
  it("renders correctly", () => {
    const { container } = render(<ProductHeader />, { wrapper: MemoryRouter });

    expect(container).toMatchSnapshot();
  });
});
