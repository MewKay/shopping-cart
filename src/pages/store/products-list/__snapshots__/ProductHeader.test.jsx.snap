// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductHeader > renders correctly 1`] = `
<div>
  <div
    class="_products-list-header_772fc5"
  >
    <ul
      class="_category-filter_772fc5"
    >
      <li>
        <a
          class=""
          href="/store/all"
        >
          All
        </a>
      </li>
      <li>
        <a
          class=""
          href="/store/men's clothing"
        >
          Men's Clothing
        </a>
      </li>
      <li>
        <a
          class=""
          href="/store/women's clothing"
        >
          Women's Clothing
        </a>
      </li>
      <li>
        <a
          class=""
          href="/store/jewelery"
        >
          Jewelery
        </a>
      </li>
      <li>
        <a
          class=""
          href="/store/electronics"
        >
          Electronics
        </a>
      </li>
    </ul>
    <input
      class="_search-product_772fc5"
      placeholder="Search for a product..."
      type="text"
      value=""
    />
  </div>
</div>
`;
